doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,332 @@
+
+- In each guest VM configuration:
+
+  - Select ``GUEST_FLAG_NVMX_ENABLED`` on :option:`vm.guest_flags.guest_flag` on the SOS VM section
+    to enable the nested virtualization feature on the Service VM.
+  - Select ``GUEST_FLAG_LAPIC_PASSTHROUGH`` on :option:`vm.guest_flags.guest_flag` to enable local
+    APIC passthrough on the Service VM.
+
+    .. figure:: images/nvmx_cfg_3.png
+       :width: 700px
+       :align: center
+
+       Service VM (SOS) ``guest_flag`` settings
+
+  - Edit :option:`vm.cpu_affinity.pcpu_id` to assign ``pCPU`` IDs to run the Service VM. If you are
+    using debug build and need the hypervisor console, don't assign
+    ``pCPU0`` to the Service VM.
+
+    You may need to manually edit the ACRN scenario XML configuration file to edit the ``pcpu_id`` for the Service VM (SOS):
+
+    .. code-block:: xml
+       :emphasize-lines: 5,6,7
+
+       <vm id="0">
+         <vm_type>SOS_VM</vm_type>
+         <name>ACRN SOS VM</name>
+         <cpu_affinity>
+           <pcpu_id>1</pcpu_id>
+           <pcpu_id>2</pcpu_id>
+           <pcpu_id>3</pcpu_id>
+         </cpu_affinity>
+         <guest_flags>
+           <guest_flag>GUEST_FLAG_NVMX_ENABLED</guest_flag>
+           <guest_flag>GUEST_FLAG_LAPIC_PASSTHROUGH</guest_flag>
+         </guest_flags>
+
+   The Service VM's virtual legacy UART interrupt doesn't work with LAPIC
+   passthrough, which may prevent the Service VM from booting. Instead, we need to use
+   the PCI-vUART for the Service VM. Refer to :ref:`Enable vUART Configurations <vuart_config>`
+   for more details about VUART configuration.
+
+   - Set :option:`vm.legacy_vuart.base` in ``legacy_vuart 0`` to ``INVALID_LEGACY_PIO``
+
+   - Set :option:`vm.console_vuart.base` in ``console_vuart 0`` to ``PCI_VUART``
+
+     .. figure:: images/nvmx_cfg_2.png
+        :width: 500px
+        :align: center
+
+        Service VM legacy and console vUART settings
+
+- Build with the XML configuration, referring to :ref:`getting-started-building`.
+
+
+Prepare for Service VM Kernel and rootfs
+****************************************
+
+The service VM can run Ubuntu or other Linux distributions.
+Instructions on how to boot Ubuntu as the Service VM can be found in
+:ref:`rt_industry_ubuntu_setup`.
+
+The Service VM kernel needs to be built from the ``acrn-kernel`` repo, and some changes
+to the kernel ``.config`` are needed.
+
+Instructions on how to build and install the Service VM kernel can be found
+in :ref:`Build and Install the ACRN Kernel <build-and-install-ACRN-kernel>`.
+
+Here is the quick start of how to modify and build the kernel:
+
+.. code-block:: none
+
+   git clone https://github.com/projectacrn/acrn-kernel
+   cd acrn-kernel
+   git checkout master
+   cp kernel_config_uefi_sos .config
+   make olddefconfig
+
+The following configuration entries are needed to launch nested
+guests on the Service VM:
+
+.. code-block:: none
+
+   CONFIG_KVM=y
```

Comment:
A bit of a side question: could we turn these on already today by default or are there side-effect to having those?

Reply:
kernel_config_uefi_sos is for SOS which doesn't need KVM, not really any side-effect. Since nested is considered as an experimental feature, I think it's good to keep kernel_config_uefi_sos  clean.

Reply:
OK, let's keep it clean for now. We can revisit this later.